packages/wouter/src/use-hash-location.js

```diff
@@ -22,21 +22,26 @@ const subscribeToHashUpdates = (callback) => {
 // leading '#' is ignored, leading '/' is optional
 const currentHashLocation = () => "/" + location.hash.replace(/^#?\/?/, "");
 
-export const navigate = (to, { state = null } = {}) => {
-  // calling `replaceState` allows us to set the history
-  // state without creating an extra entry
+export const navigate = (to, { state = null, replace = false } = {}) => {
   const [hash, search] = to.replace(/^#?\/?/, "").split("?");
 
-  history.replaceState(
-    state,
-    "",
-    // keep the current pathname, but replace query string and hash
-    location.pathname +
-      (search ? `?${search}` : location.search) +
-      // update location hash, this will cause `hashchange` event to fire
-      // normalise the value before updating, so it's always preceeded with "#/"
-      (location.hash = `#/${hash}`)
-  );
+  const newRelativePath =
+    location.pathname + (search ? `?${search}` : location.search) + `#/${hash}`;
+  const oldURL = window.location.href;
+  const newURL = new URL(newRelativePath, window.location.origin).href;
```

Comment:
Just to make it more consistent with the rest of the code

```suggestion
  const oldURL = location.href;
  const newURL = new URL(newRelativePath, location.origin).href;
```